db/hybrid_logical_vector.go

```diff
@@ -85,7 +85,17 @@ func (hlv *HybridLogicalVector) AddVersion(newVersion CurrentVersionVector) erro
 	if hlv.PreviousVersions == nil {
 		hlv.PreviousVersions = make(map[string]uint64)
 	}
-	hlv.PreviousVersions[hlv.SourceID] = hlv.Version
+	// we need to check if source ID already exists in PV, if so we need to ensure we are only updating with the
```

Comment:
I feel like the check on line 70 should only be used when newVersion.SourceID == hlv.sourceID.  Do you agree?

Reply:
Yes I agree I think? I'm not sure I fully understand though, my worry was about protecting against us trying to add a new version from different source id with version less than current version. 

E.g. local HLV CV = src: bucket1, vrs: 123 then call this function with the above change attempting to add CV src: bucket2, vrs: 12 would success in adding the incoming CV pair to local CV then moving the bucket1, 123 pair to PV? Given that the incoming pair is less than current pair in terms of version this seems incorrect? Unless you're suggesting tis isn't needed as thus function shouldn't be called unless this incoming pair is a newer CV pair?

Reply:
An example of the above is testcase4 in my test (i've had to change the expected output on it due to the above change) TestAddNewerVersionsBetweenTwoVectors, In my mind having def:15 as CV and abc:20 in PV on expected output seems wrong as abc's version is greater than def's?

Reply:
We don't have a requirement for sources to have clock synchronization - we expect clock skew, particularly for non-server peers.